src/env/getters.ts

```diff
@@ -6,3 +8,15 @@ export const getJwtSecret = (): string => {
   }
   return process.env.JWT_SECRET || 'dev';
 };
+
+export const getJwtExpirationPeriod = (): string => {
+  if (!process.env.JWT_EXPIRATION_PERIOD_SECONDS) {
+    return '7d';
+  }
+  const expirationSeconds = Number(process.env.JWT_EXPIRATION_PERIOD_SECONDS);
+  if (Number.isNaN(expirationSeconds)) {
+    throw new Error('JWT_EXPIRATION_PERIOD_SECONDS must be a number');
+  }
+  // https://www.npmjs.com/package/jsonwebtoken for `expiresIn` format
+  return ms(expirationSeconds * 1000);
```

Comment:
The documentation specifies that it accept plain seconds OR `ms` format, why did you decide to introduce `ms` in this PR?

Reply:
1. expiration period is provided through env variable:
    - it is imo nicer to provide a string over a long seconds number
2. `jsonwebtoken` docs says that the numeric value represents number of seconds (true)
3. the packages (`ms`) that `jsonwebtoken` references claim that numeric value represents the number of milliseconds (also true, so jsonwebtoken probably does *1000 under the hood)

so for (1) - human readability and (2) + (3) inconsistency, i would prefer to pass the expiration to the jwt encoder in a string format for explicitness.

alternatively it's obviously possible to just force everything to be seconds. and accept the downside of having to manually convert + accept absence of the explicit time units.

Reply:
1. it is imo nicer to provide a string over a long seconds number for readability and debugging purposes
2. `jsonwebtoken` docs says that the numeric value represents number of seconds (true)
3. the packages (`ms`) that `jsonwebtoken` references claim that numeric value represents the number of milliseconds (also true, so jsonwebtoken probably does *1000 under the hood)

so for (1) - human readability and (2) + (3) inconsistency, i would prefer to pass the expiration to the jwt encoder in a string format for explicitness.

alternatively it's obviously possible to just force everything to be seconds. and accept the downside of having to manually convert + accept absence of the explicit time units.

Reply:
> it is imo nicer to provide a string over a long seconds number for readability and debugging purposes

Fine with me, you can accept strings from the env variable if you are able to validate them with `ms`

Reply:
> it is imo nicer to provide a string over a long seconds number for readability and debugging purposes

> for the env i first validate the env var to be a number.

Aren't those contradictory statements? 

I suggest you define `JWT_EXPIRATION_PERIOD` as
```
const JWT_EXPIRATION_PERIOD = process.env.JWT_EXPIRATION_PERIOD ?? '7d'
```
and then validate `JWT_EXPIRATION_PERIOD` no matter if it's declared in the code or supplied via env

Reply:
> Aren't those contradictory statements?

bad wording perhaps. "nicer when a programm provides you a ...". Since it's almost 0 effort to convert once when you're setting the env but ~`almost-0`*`logLength` effort when you're reading the logs where these are seconds.

> How is it known?

due to the logic present in `env/getters.ts` which rejects non-numbers.

____

adjusted the logic now. It's possible to give both `12` and `7h` which will be correct. On incorrect values it will throw.